### ISSUE TITLE
adopt envoy.extensions.wasm.v3.PluginConfig.fail_policy

### ISSUE THREAD

**Issue by Stevenjin8** [istio/istio#53902]
### Is this the right place to submit this?

- [X] This is not a security vulnerability or a crashing bug
- [X] This is not a question about how to use Istio

### Bug Description

Use `envoy.extensions.wasm.v3.PluginConfig.fail_policy` over `envoy.extensions.wasm.v3.PluginConfig.fail_open` in Istio 1.26. envoy recently deprecated the former, but the updated proto definitions didn't make it to 1.24. To maintain n-1 backcompatibility, we should ship 1.25 with the updated protos (#53898) and don't use the deprecated fields in 1.26.

### Version

```Text
-
```

**Comment by zirain** [comment]
I thinks it should be istio 1.25.
I will take a look later.

**Comment by Stevenjin8** [comment]
@zirain I don't think we can do this in Istio 1.25. We are using commit b564fa42caba77c77dc795e82f2d37e264578a33 in 1.24 proxies 

https://github.com/istio/proxy/blob/200056a2fbaff793ad7004bc66637feec91e6a96/WORKSPACE#L26

and this commit doesnt seem to have the updated protos.
